### ISSUE TITLE
client CA file can't be found on v1.22.8 capz Windows image

### ISSUE THREAD

**Issue by jackfrancis** [kubernetes-sigs/image-builder#853]
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

**What did you expect to happen:**

Since building the Kubernetes v1.22.8 image of Windows for capz, we aren't able to join nodes to the cluster. The error suggests the CA cert file location is invalid.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Logs from Windows node kubelet:

```
E0407 19:02:06.931339    4276 server.go:271] "Failed to construct kubelet dependencies" err="unable to load client CA file C:\\var\\lib\\kubelet\\etc\\kubernetes\\pki\\ca.crt: open C:\\var\\lib\\kubelet\\etc\\kubernetes\\pki\\ca.crt: The system cannot find the path specified."
```

**Environment:**

Project ([Image Builder for Cluster API](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi), [kube-deploy/imagebuilder](https://github.com/kubernetes-sigs/image-builder/tree/master/images/kube-deploy/imagebuilder), [konfigadm](https://github.com/kubernetes-sigs/image-builder/tree/master/images/konfigadm)):

Additional info for Image Builder for Cluster API related issues:

- OS (e.g. from `/etc/os-release`, or `cmd /c ver`):
- Packer Version:
- Packer Provider:
- Ansible Version: 
- Cluster-api version (if using):
- Kubernetes version: (use `kubectl version`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by jackfrancis** [comment]
Here are logs from a repro:

https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2209/pull-cluster-api-provider-azure-e2e/1512133680686960640/artifacts/clusters/capz-e2e-xue983-ha/machines/capz-e2e-xue983-ha-md-win-ff5c5ffb9-45nnk/

The originating test:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2209/pull-cluster-api-provider-azure-e2e/1512133680686960640

**Comment by jsturtevant** [comment]
This looks related to the change in https://github.com/kubernetes-sigs/image-builder/pull/785

**Comment by jsturtevant** [comment]
cc @knabben

**Comment by jsturtevant** [comment]
This doesn't seem to occur on a 1.23 cluster that was built with these changes.  The validation cluster I ran against for this change was 1.22 as well https://github.com/kubernetes-sigs/image-builder/pull/785#issuecomment-1054694577

**Comment by knabben** [comment]
@jackfrancis do you have an existent old config loaded for this cluster? I think this is coming from the kubelet dynamicconfig

**Comment by CecileRobertMichon** [comment]
this was from a CAPZ CI test, it's repro'ing on all new 1.22.8 clusters built from CI

**Comment by jsturtevant** [comment]
So it seems something changed between 1.23 and 1.22 though I haven't been able to track it down yet. 

This was a known problem: something changed between 1.22 and 1.23 https://github.com/kubernetes/kubernetes/pull/77710#issuecomment-491994862

With capz `1dot23dot5-windows` built with the change on March 29th this cluster provisions.

**Comment by knabben** [comment]
On 1.23 we have a `mutateStringField` function to avoid the issue.

 https://github.com/kubernetes/kubernetes/blob/v1.23.5/cmd/kubeadm/app/componentconfigs/kubelet.go#L270

**Comment by jsturtevant** [comment]
Great find! The fix came in early in 1.23 https://github.com/kubernetes/kubernetes/pull/105992

I propose we do the symlink on versions < 1.23 as the fix.
